### PR TITLE
Surprise for PM

### PR DESCRIPTION
**Pull request by skateman**
Had to do some cleanups and extend some things around `CERTH_AUTH` which allowed me to add additional test coverage for some endpoints. Furthermore, I introduced searching profiles by `os_minor_version` to allow insights-client to look up profiles to which a system can be assigned.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -19,9 +19,8 @@ def authorized_to?(access_request)
 
   def inventory_groups
     # No need to fetch inventory groups if the RBAC feature is globally disabled
-    return Rbac::ANY if rbac_disabled?
-    # No groups should be available when using certificate authentication
-    return [] if cert_authenticated?
+    # All groups should be available when using certificate authentication
+    return Rbac::ANY if rbac_disabled? || cert_authenticated?
```
Comment (vkrizan) [inline app/models/user.rb:23]: Authenticated requests with certificate would have access to everything regardless of group restrictions. Is is possible to tie the certificate to a specific system?

**Review by vkrizan — COMMENTED** [review]
Code-wise looks good.

My main concern is with permissive access when using certificate.

File: app/models/user.rb (hunk 1 above)
Comment (skateman) [reply app/models/user.rb:23]: Valid point, found some stuff to read in the inventory README about this...will experiment a bit.

Comment (skateman) [reply app/models/user.rb]: @vkrizan fixed

File: app/policies/v2/system_policy.rb [hunk 2]
```diff
@@ -38,18 +38,27 @@ def match_group?
...
 
-        user_scope = scope.where(org_id: user.org_id)
+        # Apply inventory group rules on the query if needed
+        groups == Rbac::ANY ? base_scope : base_scope.with_groups(groups)
+      end
 
-        # All systems are available if there is global access
-        return user_scope if groups == Rbac::ANY
+      def resolve_cert_auth
+        base_scope.where(V2::System::OWNER_ID.eq(user.system_owner_id))
```
Comment (vkrizan) [inline app/policies/v2/system_policy.rb:57]: Note to QE: This should be thoroughly tested to see whether all systems can run insights client without any issues.

File: app/controllers/concerns/v2/authentication.rb [hunk 3]
```diff
@@ -7,6 +7,8 @@ module Authentication
 
     ALLOWED_CERT_BASED_RBAC_ACTIONS = [
       { controller: 'policies', action: 'index', parents: %i[systems] },
+      { controller: 'systems', action: 'update', parents: %i[policies] },
+      { controller: 'systems', action: 'destroy', parents: %i[policies] },
```
Comment (vkrizan) [inline app/controllers/concerns/v2/authentication.rb:12]: Are these operations restricted only to that single, cert-authd, system?

**Review by vkrizan — APPROVED** [review]

File: app/controllers/concerns/v2/authentication.rb (hunk 3 above)
Comment (skateman) [reply app/controllers/concerns/v2/authentication.rb:12]: Yes, check the tests

Comment (skateman) [reply app/controllers/concerns/v2/authentication.rb:12]: https://github.com/RedHatInsights/compliance-backend/pull/2029/files#diff-42f377e9eff99098da9940c7d84cc92a4db807694709b5a14164dd831f4cf986R517-R524

and

https://github.com/RedHatInsights/compliance-backend/pull/2029/files#diff-42f377e9eff99098da9940c7d84cc92a4db807694709b5a14164dd831f4cf986R566-R574

**Pull request merged**